crypto/src/key/secp256k1/extended_keys.rs

```diff
@@ -157,12 +178,56 @@ impl Secp256k1ExtendedPublicKey {
 
     pub fn from_private_key(private_key: &Secp256k1ExtendedPrivateKey) -> Self {
         Secp256k1ExtendedPublicKey {
-            public_key: private_key.private_key.data.x_only_public_key(SECP256K1).0.into(),
+            derivation_path: private_key.derivation_path.clone(),
             chain_code: private_key.chain_code,
+            public_key: private_key.private_key.data.public_key(SECP256K1).into(),
         }
     }
 }
 
+impl Derivable for Secp256k1ExtendedPublicKey {
+    fn derive_child(self, num: ChildNumber) -> Result<Self, DerivationError> {
+        // Create a new hmac with the chain code as the key
+        let mut mac = new_hmac_sha_512(&self.chain_code.into_array());
+
+        if num.is_normal() {
+            mac.update(&self.public_key.pubkey_data.serialize());
+        } else {
+            return Err(DerivationError::CannotDeriveHardenedKeyFromPublicKey(num));
+        }
+
+        // Add the child number
+        mac.update(&num.into_encoded_be_bytes());
+
+        // Finalize and get the new un-tweaked key and the new chain code
+        let (key_part, chain_code) = to_key_and_chain_code(mac)?;
+
+        // Create the derived public key
+        let public_key = self
+            .public_key
+            .pubkey_data
+            .add_exp_tweak(SECP256K1, &key_part.into())
+            .map_err(|_| DerivationError::KeyDerivationError)?
+            .into();
+
+        let derivation_path = {
+            let mut child_path = self.get_derivation_path().into_vec();
+            child_path.push(num);
+            child_path.try_into()?
```

Comment:
Originally I wanted `get_derivation_path` to return a reference but had issues because of `MLRistrettoPrivateKey` that we will remove soon. So I did some work that makes this method return a reference.